### PR TITLE
Updated to use current version of Horreum 0.15.x

### LINKED ISSUE #17: Version bump horreum-plugin to use current 0.15.x Horreum version.
### What feature do you want to see added?

Use a current version of Horreum.

### Upstream changes

https://github.com/Hyperfoil/Horreum/releases/tag/0.15.0

### Are you interested in contributing this feature?

Yes.

### PR DESCRIPTION
**Pull request by whitingjr**
Fixes #17 feature request.

This PR updates the plugin to use the most recent version of Horreum libraries and api.

### Testing done

 Used existing JUnit test cases to ensure the operations provided are functioning as expected after the version bump.

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue
- [x] Replace SNAPSHOT dependency with released library version

### CODE REVIEW

**Comment by whitingjr** [comment]
I have observed the creating of a HorreumClient object is flakey due to `Connection reset` exceptions. Retries have been included to solve this. In the HorreumTestClientExtension. This might still be un-reliable when running on machines that are slower. I am thinking a better solution is to attempt creating a client for a minute after the Horreum container starts.

**Comment by johnaohara** [comment]
> I have observed the creating of a HorreumClient object is flakey due to `Connection reset` exceptions. Retries have been included to solve this. In the HorreumTestClientExtension. This might still be un-reliable when running on machines that are slower. I am thinking a better solution is to attempt creating a client for a minute after the Horreum container starts.

@whitingjr I pinged you offline, but looks like that crossed with this comment :). I will copy here;

There is a HorreumResources.waitForContainerReady() method that is used to wait until the containers are ready before progressing, you can see it used in HorreumResources.startContainers() .  This scans the container log, looking for a particular string, and throws an exception after a timeout.  TestContainers also has a number of different waitFor strategies for pausing execution until different conditions are met if scanning the log is not sufficient. 

This method can be used to ensure the services are started before trying to instantiate the client, and remove the need for any retry logic

**Comment by whitingjr** [comment]
Thanks for pointing these out. I'll replace the retry and update the branch.

**Comment by johnaohara** [comment]
@whitingjr I have opened a PR in Horreum that removes all of the Quarkus transitive dependencies in `horreum-infra-common`: https://github.com/Hyperfoil/Horreum/pull/2017

They are not needed and will greatly simplify the transitive dependency management of the plugin.

The main difference with this change is  "quarkus.http.port" & "quarkus.http.host" need to be added to the map passed to `HorreumResources.startContainers()` : https://github.com/Hyperfoil/Horreum/pull/2017/files#diff-b43105658cc21b6fa970980abf2234869cdc5fed60280eae9d557a6b89ed54daR112-R116

**Comment by whitingjr** [comment]
Both properties are now added and the Quarkus transitive dependencies are no longer part of the project.

File: src/test/java/jenkins/plugins/horreum/junit/HorreumTestExtension.java
Comment (johnaohara) [inline src/test/java/jenkins/plugins/horreum/junit/HorreumTestExtension.java:49]: this should be loaded from `META-INF/horreum.dev.properties` published in `horreum-infra-common` module, after this PR has been merged: https://github.com/Hyperfoil/Horreum/pull/2023

**Review by johnaohara — CHANGES_REQUESTED** [review]
I think the tests and container services look great! I think the dependencies should be cleaned up aligned to latest LTS for Jenkins, comments are inline in POM

I have also opened this PR in Horreum to simplify the container image management: https://github.com/Hyperfoil/Horreum/pull/2023

File: README.adoc
Comment (whitingjr) [reply README.adoc:74]: My understanding is hpi:run is to stand up Jenkins as you said. But it won't initiate the tests.

File: pom.xml
Comment (whitingjr) [reply pom.xml:69]: sure.

File: pom.xml
Comment (whitingjr) [reply pom.xml]: sure

File: src/test/java/jenkins/plugins/horreum/junit/HorreumTestExtension.java
Comment (whitingjr) [reply src/test/java/jenkins/plugins/horreum/junit/HorreumTestExtension.java:49]: updated to load from the properties file.

**Comment by whitingjr** [comment]
@johnaohara  For me when running the test-suite a 2nd time establishing a HorreumClient object causes `Connection reset`.
```
$ mvn clean test;mvn test
.....
[ERROR] Errors: 
[ERROR]   HorreumExpectStepTest.testExpect:22->HorreumPluginTestBase.createTest:18 » Runtime java.net.SocketException: Connection reset
[ERROR]   HorreumExpectTest.testExpectRun:22->HorreumPluginTestBase.createTest:18 » Runtime java.net.SocketException: Connection reset
[ERROR]   HorreumUploadStepTest.testUpload:23->HorreumPluginTestBase.createTest:18 » Runtime java.net.SocketException: Connection reset
[ERROR]   HorreumUploadStepTest.testUploadMultiple:59->HorreumPluginTestBase.createTest:18 » Runtime java.net.SocketException: Connection reset
[ERROR]   HorreumUploadTest.testUpload:32->HorreumPluginTestBase.createTest:18 » Runtime java.net.SocketException: Connection reset
[ERROR]   HorreumUploadTest.testUploadMultiple:55->HorreumPluginTestBase.createTest:18 » Runtime java.net.SocketException: Connection reset
[INFO] 
[ERROR] Tests run: 6, Failures: 0, Errors: 6, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```
 In the first execution of the test goal jenkins-for-test being unpacked for booting the Jetty server causes sufficient delay for HorreumClient to authenticate. The second test goal uses the existing unpacked directory which is quicker. 
 Can you recreate this issue?

**Comment by johnaohara** [comment]
> There is one weakness of the test cases being separate objects. The test-suite is not able to use AfterAll annotation. It is executed for every test class which causes containers to be recycled. Docker handles shutting down running containers when the tests all cease but Podman I don't recall. Was there a reason for `HorreumUploadStepTest`, `HorreumUploadTest`, `HorreumExpectStepTest`, `HorreumExpectTest` etc ?

I don't know the answer to this question

**Comment by johnaohara** [comment]
> @johnaohara For me when running the test-suite a 2nd time establishing a HorreumClient object causes `Connection reset`.
> 
> ```
> $ mvn clean test;mvn test
> .....
> [ERROR] Errors: 
> [ERROR]   HorreumExpectStepTest.testExpect:22->HorreumPluginTestBase.createTest:18 » Runtime java.net.SocketException: Connection reset
> [ERROR]   HorreumExpectTest.testExpectRun:22->HorreumPluginTestBase.createTest:18 » Runtime java.net.SocketException: Connection reset
> [ERROR]   HorreumUploadStepTest.testUpload:23->HorreumPluginTestBase.createTest:18 » Runtime java.net.SocketException: Connection reset
> [ERROR]   HorreumUploadStepTest.testUploadMultiple:59->HorreumPluginTestBase.createTest:18 » Runtime java.net.SocketException: Connection reset
> [ERROR]   HorreumUploadTest.testUpload:32->HorreumPluginTestBase.createTest:18 » Runtime java.net.SocketException: Connection reset
> [ERROR]   HorreumUploadTest.testUploadMultiple:55->HorreumPluginTestBase.createTest:18 » Runtime java.net.SocketException: Connection reset
> [INFO] 
> [ERROR] Tests run: 6, Failures: 0, Errors: 6, Skipped: 0
> [INFO] 
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD FAILURE
> [INFO] ------------------------------------------------------------------------
> ```
> 
> In the first execution of the test goal jenkins-for-test being unpacked for booting the Jetty server causes sufficient delay for HorreumClient to authenticate. The second test goal uses the existing unpacked directory which is quicker. Can you recreate this issue?

do we need to do this?

**Comment by whitingjr** [comment]
The CI is rejecting the build for jdk8 and jdk11.
It seems we are too ambitious targeting 17. Unless the build pipeline is updated.
https://ci.jenkins.io/job/Plugins/job/horreum-plugin/job/PR-18/

**Comment by johnaohara** [comment]
> The CI is failing the build for jdk8 and jdk11. It seems we are too ambitious targeting 17. Unless the build pipeline is updated. https://ci.jenkins.io/job/Plugins/job/horreum-plugin/job/PR-18/

the project will need to align to the minimum version that the LTS jenkins version supports

**Comment by whitingjr** [comment]
> the project will need to align to the minimum version that the LTS jenkins version supports

That is 11 in plugin-pom/pom.xml for [4.85](https://github.com/jenkinsci/plugin-pom/blob/plugin-4.85/pom.xml#L61) tag.
Setting the target to 11 in horreum-plugin/pom.xml then causes this enforcer rule violation.
```
[INFO] --- enforcer:3.1.0:enforce (display-info) @ horreum ---
[INFO] Restricted to JDK 11 yet io.hyperfoil.tools:horreum-client:jar:0.15.1:compile contains io/hyperfoil/tools/RunServiceExtension.class targeted to JDK 17
[INFO] Restricted to JDK 11 yet io.hyperfoil.tools:horreum-api:jar:0.15.1:compile contains io/hyperfoil/tools/horreum/api/ApiIgnore.class targeted to JDK 17
[WARNING] Unknown bytecodeVersion for com.fasterxml.jackson.core:jackson-core:jar:2.17.2:compile : META-INF/versions/21/com/fasterxml/jackson/core/io/doubleparser/FastDoubleSwar.class: got null class-file-version
[WARNING] Unknown bytecodeVersion for com.fasterxml.jackson.core:jackson-core:jar:2.17.2:compile : META-INF/versions/21/com/fasterxml/jackson/core/io/doubleparser/FastIntegerMath.class: got null class-file-version
[ERROR] Rule 3: org.apache.maven.plugins.enforcer.EnforceBytecodeVersion failed with message:
Found Banned Dependency: io.hyperfoil.tools:horreum-client:jar:0.15.1
Found Banned Dependency: io.hyperfoil.tools:horreum-api:jar:0.15.1
```

Is there any prospect of getting a JDK 11 compile targeted version for the 2 modules ?

**Comment by johnaohara** [comment]
> > the project will need to align to the minimum version that the LTS jenkins version supports
> 
> That is 11 in plugin-pom/pom.xml for [4.85](https://github.com/jenkinsci/plugin-pom/blob/plugin-4.85/pom.xml#L61) tag. Setting the target to 11 in horreum-plugin/pom.xml then causes this enforcer rule violation.
> 
> ```
> [INFO] --- enforcer:3.1.0:enforce (display-info) @ horreum ---
> [INFO] Restricted to JDK 11 yet io.hyperfoil.tools:horreum-client:jar:0.15.1:compile contains io/hyperfoil/tools/RunServiceExtension.class targeted to JDK 17
> [INFO] Restricted to JDK 11 yet io.hyperfoil.tools:horreum-api:jar:0.15.1:compile contains io/hyperfoil/tools/horreum/api/ApiIgnore.class targeted to JDK 17
> [WARNING] Unknown bytecodeVersion for com.fasterxml.jackson.core:jackson-core:jar:2.17.2:compile : META-INF/versions/21/com/fasterxml/jackson/core/io/doubleparser/FastDoubleSwar.class: got null class-file-version
> [WARNING] Unknown bytecodeVersion for com.fasterxml.jackson.core:jackson-core:jar:2.17.2:compile : META-INF/versions/21/com/fasterxml/jackson/core/io/doubleparser/FastIntegerMath.class: got null class-file-version
> [ERROR] Rule 3: org.apache.maven.plugins.enforcer.EnforceBytecodeVersion failed with message:
> Found Banned Dependency: io.hyperfoil.tools:horreum-client:jar:0.15.1
> Found Banned Dependency: io.hyperfoil.tools:horreum-api:jar:0.15.1
> ```
> 
> Is there any prospect of getting a JDK 11 compile targeted version for the 2 modules ?

They *should* be targeting `1.8`, but looks like the target definition is wrong in the poms: 

https://github.com/Hyperfoil/Horreum/blob/0.15/horreum-api/pom.xml#L13-L14

https://github.com/Hyperfoil/Horreum/blob/0.15/horreum-client/pom.xml#L18-L19

**Comment by johnaohara** [comment]
@whitingjr : Fix target version for `horreum-api` and `horreum-client` https://github.com/Hyperfoil/Horreum/pull/2026

**Comment by whitingjr** [comment]
> They _should_ be targeting `1.8`, but looks like the target definition is wrong in the poms:
> 
> https://github.com/Hyperfoil/Horreum/blob/0.15/horreum-api/pom.xml#L13-L14
> 
> https://github.com/Hyperfoil/Horreum/blob/0.15/horreum-client/pom.xml#L18-L19

Yes it seems to be targeted to 17.

```
$ javap -v horreum-api/target/classes/io/hyperfoil/tools/horreum/api/Version.class |grep major
  major version: 61
```

The effective pom reveals this for horreum-api
```
<maven.compiler.release>17</maven.compiler.release>
    <maven.compiler.source>8</maven.compiler.source>
    <maven.compiler.target>8</maven.compiler.target>
```

by adding the `release` property to horreum-api/pom.xml for the compiler plugin it creates the targeted version correctly.

```
$ javap -v horreum-api/target/classes/io/hyperfoil/tools/horreum/api/Version.class |grep major
  major version: 52
```
Shall I create a PR ?

**Comment by johnaohara** [comment]
A PR is already merged: https://github.com/Hyperfoil/Horreum/pull/2026

**Comment by johnaohara** [comment]
@whitingjr horreum `0.15.2` has been released that includes the fix for class versions

**Comment by whitingjr** [comment]
@johnaohara I am seeing horreum-infra-common now causing an issue.

**Comment by johnaohara** [comment]
> @johnaohara I am seeing horreum-infra-common now causing an issue.
> 
> ```
> ERROR]   bad class file: /home/whitingjr/.m2/repository/io/hyperfoil/tools/horreum-infra-common/0.15.2/horreum-infra-common-0.15.2.jar(/io/hyperfoil/tools/horreum/infra/common/SelfSignedCert.class)
> [ERROR]     class file has wrong version 61.0, should be 55.0
> ```
> 
> A test locally with horreum-infra-common module compiled to 11 has solved all the issues.

Is that running the build with JDK 11? we don't need to run the build/tests with JDK 11, we can use JDK17+, but release should *target* jdk11

**Comment by whitingjr** [comment]
> Is that running the build with JDK 11? we don't need to run the build/tests with JDK 11, we can use JDK17+, but release should _target_ jdk11

```
		<maven.compiler.target>17</maven.compiler.target>
		<maven.compiler.source>17</maven.compiler.source>
		<maven.compiler.release>11</maven.compiler.release>
```
 compiling with the build/test to 17 and release at 11 the error still occurs.

**Comment by johnaohara** [comment]
> > Is that running the build with JDK 11? we don't need to run the build/tests with JDK 11, we can use JDK17+, but release should _target_ jdk11
> 
> ```
> 		<maven.compiler.target>17</maven.compiler.target>
> 		<maven.compiler.source>17</maven.compiler.source>
> 		<maven.compiler.release>11</maven.compiler.release>
> ```
> 
> compiling with the build/test to 17 and release at 11 the error still occurs.

what version of the JDK are you using to build with?

**Comment by whitingjr** [comment]
Good catch. I was using 11. With 17 it is ok.

**Comment by whitingjr** [comment]
> > There is one weakness of the test cases being separate objects. The test-suite is not able to use AfterAll annotation. It is executed for every test class which causes containers to be recycled. Docker handles shutting down running containers when the tests all cease but Podman I don't recall. Was there a reason for `HorreumUploadStepTest`, `HorreumUploadTest`, `HorreumExpectStepTest`, `HorreumExpectTest` etc ?
> 
> I don't know the answer to this question

 To get the advantage of using AfterAll annotation is there any reason not to combine these separate objects into a single object ?

**Comment by whitingjr** [comment]
@johnaohara  I have updated Jenkinsfile with [configuration to change the CI builds](https://github.com/whitingjr/horreum-plugin/blob/horreum-plugin-version-bump-horreum-to-0.14-v2/Jenkinsfile#L3-L13). This isn't incorporated automatically to avoid malicious changes. Is there an alternative process to reconfigure the CI ?

**Comment by johnaohara** [comment]
> > > There is one weakness of the test cases being separate objects. The test-suite is not able to use AfterAll annotation. It is executed for every test class which causes containers to be recycled. Docker handles shutting down running containers when the tests all cease but Podman I don't recall. Was there a reason for `HorreumUploadStepTest`, `HorreumUploadTest`, `HorreumExpectStepTest`, `HorreumExpectTest` etc ?
> > 
> > 
> > I don't know the answer to this question
> 
> To get the advantage of using AfterAll annotation is there any reason not to combine these separate objects into a single object ?

I don't see why not

**Comment by johnaohara** [comment]
> @johnaohara I have updated Jenkinsfile with [configuration to change the CI builds](https://github.com/whitingjr/horreum-plugin/blob/horreum-plugin-version-bump-horreum-to-0.14-v2/Jenkinsfile#L3-L13). This isn't incorporated automatically to avoid malicious changes. Is there an alternative process to reconfigure the CI ?

I do not know how CI/CD works for this project, that is something that needs figuring out

**Comment by whitingjr** [comment]
> I do not know how CI/CD works for this project, that is something that needs figuring out

I attempted to introduce a minimal commit to only Jenkinsfile to remove jdk8. That's failed due to enforcer not finding a dependency
```
[2024-09-25T14:11:25.178Z] [WARNING] Rule 5: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:
[2024-09-25T14:11:25.178Z] Could not build dependency tree Could not collect dependencies: io.jenkins.plugins:horreum:hpi:0.1.551.v72b_640997d25 ()
[2024-09-25T14:11:25.178Z] [INFO] ------------------------------------------------------------------------
[2024-09-25T14:11:25.179Z] [INFO] BUILD FAILURE
[2024-09-25T14:11:25.179Z] [INFO] ------------------------------------------------------------------------
```
https://repo.jenkins-ci.org/artifactory/releases/io/jenkins/plugins/horreum/
Should this be published already ?

**Comment by johnaohara** [comment]
I think the build error occurred before that:


```
[2024-09-25T13:41:14.849Z] [INFO] --- hpi:3.27:validate-hpi (default-validate-hpi) @ horreum ---
[2024-09-25T13:41:15.150Z] [WARNING] Error resolving project artifact: The following artifacts could not be resolved: com.sun:tools:pom:1.8.0 (absent): Could not find artifact com.sun:tools:pom:1.8.0 in azure-internal (http://artifact-caching-proxy.privatelink.azurecr.io:8080/) for project com.sun:tools:jar:1.8.0
[2024-09-25T13:41:15.552Z] [INFO] 
[2024-09-25T13:41:15.552Z] [INFO] --- enforcer:3.0.0:display-info (display-info) @ horreum ---
[2024-09-25T13:41:17.556Z] [INFO] Maven Version: 3.9.9
[2024-09-25T13:41:17.557Z] [INFO] JDK Version: 1.8.0_422 normalized as: 1.8.0-422
[2024-09-25T13:41:17.557Z] [INFO] Java Vendor: Temurin
[2024-09-25T13:41:17.557Z] [INFO] OS Info: Arch: amd64 Family: unix Name: linux Version: 6.8.0-1014-azure
[2024-09-25T13:41:17.558Z] [INFO] 
[2024-09-25T13:41:17.558Z] [INFO] --- enforcer:3.0.0:enforce (display-info) @ horreum ---
[2024-09-25T13:41:17.661Z] [WARNING] System artifact: com.sun:tools:jar:1.8.0:system has no file attached
[2024-09-25T13:41:17.662Z] 
[2024-09-25T13:41:17.663Z] Try downloading the file manually from the project website.
[2024-09-25T13:41:17.664Z] 
[2024-09-25T13:41:17.664Z] Then, install it using the command: 
[2024-09-25T13:41:17.664Z]     mvn install:install-file -DgroupId=com.sun -DartifactId=tools -Dversion=1.8.0 -Dpackaging=jar -Dfile=/path/to/file
[2024-09-25T13:41:17.664Z] 
[2024-09-25T13:41:17.666Z] Alternatively, if you host your own repository you can deploy the file there: 
[2024-09-25T13:41:17.670Z]     mvn deploy:deploy-file -DgroupId=com.sun -DartifactId=tools -Dversion=1.8.0 -Dpackaging=jar -Dfile=/path/to/file -Durl=[url] -DrepositoryId=[id]
[2024-09-25T13:41:17.671Z] 
[2024-09-25T13:41:17.671Z] Path to dependency: 
[2024-09-25T13:41:17.671Z] 	1) com.sun:tools:jar:1.8.0
[2024-09-25T13:41:17.671Z] 
[2024-09-25T13:41:17.671Z] 
[2024-09-25T13:41:17.672Z]   com.sun:tools:jar:1.8.0
```

**Comment by whitingjr** [comment]
This PR removes JDK8 from the list of configurations to run in the CI. What's puzzling is I expect the  CI to exclude an attempt to build with JDK8. Is there an alternate place the CI configures which JDK's to execute a step for ?

File: pom.xml
Comment (whitingjr) [reply pom.xml]: For 4.52 the class files created for org.jenkins-ci.tools:maven-hpi-plugin have been created beyond JDK11. The latest version targeting JDK11 is 4.40. Unless we re-evaluate and target JDK17 for horreum-plugin for this release I will need to use 4.40.

Comment (whitingjr) [reply pom.xml:30]: 4.52 is the highest version that uses org.jenkins-ci.tools:maven-hpi-plugin:3.37 which targets JDK11 class file version. 4.53 has JDK17 class files in org.jenkins-ci.tools:maven-hpi-plugin:3.38.

File: pom.xml
Comment (whitingjr) [reply pom.xml]: Sure.

File: pom.xml
Comment (whitingjr) [reply pom.xml]: This has been updated.

File: pom.xml
Comment (whitingjr) [reply pom.xml]: Removed.

File: pom.xml
Comment (whitingjr) [reply pom.xml:285]: Without it the testsuite fails to startup. The PostgresResource fails to configure the database.  `SQLException: No suitable driver found for jdbc:postgresql://localhost:32847/horreum?loggerLevel=OFF`

Comment (johnaohara) [reply pom.xml:266]: OK, i think that dependency should be a transitive dep in `horreum-infra-common`, but that does not appear to be the case.  So lets leave this for now, and fix `horreum-infra-common` at some later stage

**Comment by johnaohara** [comment]
Hi @MarkEWaite You re-ran a PR build #19 with an updated Jenkinsfile in ci.jenkins.org.  Thank you

That PR was intended to allow this PR to run in CI on the correct target environment, but the changes are also included in this PR.

Is it possible to run with the update Jenkinsfile in this PR on `ci.jenkins.org`? is that something that I can trigger as a plugin maintainer?

**Comment by whitingjr** [comment]
@johnaohara ok I'll back out any changes in Jenkinsfile so that this PR is not involved in changes included in #20.

**Comment by johnaohara** [comment]
@whitingjr can you wait before making changes to the Jenkinsfile in this PR? The pom dependencies have been aligned against the jenkins version in the jenkinsfile in this PR. #20 has a different Jenkins version.

If we can re-run CI with the Jenkinsfile in this PR, and it passes, I think we can merge this and not need #20.  If #20 is merged, CI will likely to run into issues again because this PR will run against jenkins '2.332.2', which will likely not work

**Comment by whitingjr** [comment]
@johnaohara just seen your comment after pushing. I'll update with another commit as you suggest.

**Comment by whitingjr** [comment]
@johnaohara pushed the commit and the CI failed. JDK8 is not compatible with 2.440.3. Over on #20 the CI checks passed.

**Comment by johnaohara** [comment]
@whitingjr the re-introduced `Jenkinsfile` in https://github.com/jenkinsci/horreum-plugin/commit/5d6bdc11cc0cbe746ddda82e72ebb07d5a5736b2 is not the same as the version that was backed.

This was the backed out file: https://github.com/jenkinsci/horreum-plugin/blob/5096de19097817c9cc45f11a87933ed1b9d91f70/Jenkinsfile

This is the file that has been re-introduced: https://github.com/jenkinsci/horreum-plugin/blob/5d6bdc11cc0cbe746ddda82e72ebb07d5a5736b2/Jenkinsfile

**Comment by johnaohara** [comment]
> @johnaohara pushed the commit and the CI failed. JDK8 is not compatible with 2.440.3. Over on #20 the CI checks passed.

PR CI pipelines are defined from the current `Jenkinsfile`: https://github.com/jenkinsci/horreum-plugin/blob/9052b35210ea81915d4eb0a802a8e0f3d2f5c0b7/Jenkinsfile

This PR will not build against JDK1.8 (we have incompatible class files) and chose to target JDK11 as a minimum: this is why the CI is currently failing.

#20 remove JDK1.8, but still using jenkins: `2.332.2`.  This PR has been developed against jenkins `2.440.3`.  It *might* work against jenkins `2.332.2`, but that has not been tested, and we want to validate CI against `2.440.3`.

This was the original Jenkinsfile in this PR: https://github.com/jenkinsci/horreum-plugin/blob/5096de19097817c9cc45f11a87933ed1b9d91f70/Jenkinsfile

That version ^, removed JDK1.8, and bumped Jenkins to `2.440.3`.  This is the env that we need to run the CI tests against. 

In PR #19 Gounthar kindly commented that ` One of the maintainers would need to replay the build explicitly using your Jenkinsfile.` : https://github.com/jenkinsci/horreum-plugin/pull/19#issuecomment-2377232225

Before PR #20 was closed, [MarkEWaite](https://github.com/MarkEWaite) had re-run the CI pipeline for that PR: https://ci.jenkins.io/job/Plugins/job/horreum-plugin/view/change-requests/job/PR-19/2/  . It only ran on JDK11, but it failed because it was trying to build against `horreum:0.7` on `jenkins:2.440.3`

What needs to happen is this PR #18 needs the Orginal Jenkinsfile re-instating : https://github.com/jenkinsci/horreum-plugin/blob/5096de19097817c9cc45f11a87933ed1b9d91f70/Jenkinsfile

The CI then needs to be manually re-ran with that Jenkinsfile.  Hopefully it then passes

I have asked if I am able to re-run the CI with the updated Jenkinsfile, but is suspect not due to security restrictions: https://github.com/jenkinsci/horreum-plugin/pull/18#issuecomment-2377337197

Once the CI has passed with the Jenkinsfile manually updated from this PR, we can merge this PR and #20 is redundant.

**Comment by MarkEWaite** [comment]
Pull requests from maintainers are allowed to run the updated Jenkinsfile on ci.jenkins.io.  Pull requests from non-maintainer only use the Jenkinsfile from the target branch.

File: pom.xml
Comment (MarkEWaite) [inline pom.xml:74]: Forcing Java 17 means that users running Java 11 on Jenkins 2.440.3, 2.452.x, and 2.462.x will fail to run the plugin.

If you must force Java 17, then you should probably set the base Jenkins version to Jenkins 2.477 weekly or newer.  Jenkins LTS core will require Java 17 or newer with the 30 Oct 2024 release of Jenkins 2.477.1.
```suggestion
```

Comment (johnaohara) [reply pom.xml:74]: @MarkEWaite Thanks for spotting this!

@whitingjr the plugin should target Java 11

**Comment by johnaohara** [comment]
> Pull requests from maintainers are allowed to run the updated Jenkinsfile on ci.jenkins.io. Pull requests from non-maintainer only use the Jenkinsfile from the target branch.
> 
> I've restarted the job on ci.jenkins.io with the updated Jenkinsfile. It failed to compile.

@MarkEWaite Thank you for re-running the CI job. I am assuming that any new pushes to this PR will trigger the updated job?

@whitingjr there is a 0.15.3 release running now that will fix the compilation issue

**Comment by MarkEWaite** [comment]
> @MarkEWaite Thank you for re-running the CI job. I am assuming that any new pushes to this PR will trigger the updated job?

No, a maintainer needs to be the one that creates the pull request.  @whitingjr does not seem to be a maintainer.

**Comment by johnaohara** [comment]
> maintainer

Ok, thank you for clarifying

File: pom.xml
Comment (whitingjr) [reply pom.xml]: @johnaohara it now targets 11 and uses 0.15.3 release of Horreum artifacts.

**Review by johnaohara — APPROVED** [review]

**Comment by johnaohara** [comment]
:tada:

**Comment by johnaohara** [comment]
@MarkEWaite thank you for all your help!

**Comment by whitingjr** [comment]
@MarkEWaite thank you for your expert guidance. And everyone else who contributed.

**Pull request merged**
